docs/proposals/resource_management.md

```diff
@@ -0,0 +1,381 @@
+                limits:
+                    cpu: 10m
+                    memory: 1Gi
+        name: bar
+            resources:
+                limits:
+                    cpu: 100m
+                    memory: 2Gi
+```
+
+```yaml
+kind: Pod
+metadata:
+    name: Pod2
+spec:
+    containers:
+        name: foo
+            resources:
+                limits:
+                    cpu: 20m
+                    memory: 2Gii
+```
+
+Pod1 and Pod2 are both classified as Guaranteed and are nested under the `Root` cgroup.  
+
+/Pod1/cpu.quota = 110m  
+/Pod1/cpu.shares = 110m  
+/Pod2/cpu.quota = 20m  
+/Pod2/cpu.shares = 20m  
+/Pod1/m.limit = 3Gi  
+/Pod2/m.limit = 2Gi
+
+#### Burstable
+
+We have the following resource parameters for the `Bu` cgroup.
+
+/Bu/cpu.shares = summation of all cpu requests  
+/Bu/PodUID/cpu.quota = Pod Cpu Limit  
+/Bu/PodUID/cpu.shares = Pod Cpu Request   
+/Bu/m.limit_in_bytes = Allocatable - summation of memory requests/limits of guaranteed pods  
+/Bu/PodUID/m.limit = Pod memory limit
+
+`Note: For the Burstable Qos when limits are not specified for any one of the containers, the Pod limit defaults to the node resource capacity.`
+
+Example:  
+We have two pods Pod3 and Pod4 having Pod Spec given below:  
+
+```yaml
+kind: Pod
+metadata:
+    name: Pod3
+spec:
+    containers:
+        name: foo
+            resources:
+                limits:
+                    cpu: 50m
+                    memory: 2Gi
+                requests:
+                    cpu: 20m
+                    memory: 1Gi
+        name: bar
+            resources:
+                limits:
+                    cpu: 100m
+                        memory: 1Gi
+```
+
+```yaml
+kind: Pod
+metadata:
+    name: Pod4
+spec:
+    containers:
+        name: foo
+            resources:
+                limits:
+                    cpu: 20m
+                    memory: 2Gi
+                requests:
+                    cpu: 10m
+                    memory: 1Gi  
+```
+
+Pod3 and Pod4 are both classified as Burstable and are hence nested under the Bu cgroup  
+
+/Bu/cpu.shares = 130m  
+/Bu/Pod3/cpu.quota = 150m  
+/Bu/Pod3/cpu.shares = 20m  
+/Bu/Pod4/cpu.quota = 20m  
+/Bu/Pod4/cpu.shares = 10m  
+/Bu/m.limit_in_bytes = Allocatable - 5Gi  
+/Bu/Pod3/m.limit = 3Gi  
+/Bu/Pod4/m.limit = 2Gi  
+
+#### Best Effort
+
+For pods belonging to the Best effort Qos we don't set any quota.  
+/BE/cpu.shares = 2  
+/BE/cpu.quota= not set  
+/BE/m.limit_in_bytes = Allocatable - summation of memory limits of Guaranteed and Burstable pods  
+/BE/PodUID/m.limit = no limit  
+
+Example:  
+We have a pod 'Pod5' having Pod Spec given below:  
+
+```yaml
+kind: Pod
+metadata:
+    name: Pod5
+spec:
+    containers:
+        name: foo
+            resources:
+        name: bar
+            resources:
+```
+
+Pod5 is classified as Best effort and is hence nested under the BE cgroup  
+
+/BE/cpu.shares = 2  
+/BE/cpu.quota= not set  
+/BE/m.limit_in_bytes = Allocatable - 7Gi  
```

Comment:
In this case setting the limits as  `node resource capacity - summation of memory limit/request of guaranteed pods)`
is not required as we are already setting limits on the Burstable Qos class cgroup as `node resource capacity - summation of memory limit/request of guaranteed pods)` 
So in case of memory contention the pods inside the Burstable Qos cgroup would be limited to 
`node resource capacity - summation of memory limit/request of guaranteed pods)` by the parent Burstable cgroup